### PR TITLE
Fix Login Flow Issue

### PR DESCRIPTION
**Pull request by hunyoboy**

### CODE REVIEW

**Comment by hunyoboy** [comment]
@efjacobson please review if you're free. thank you.

File: src/plugins/iam/sagas/index.js [hunk 1]
```diff
@@ -10,15 +10,20 @@ import { actionTypes, serviceIds } from '../constants';
...
  */
 function* watchLogin(container) {
+  let lastTask;
   while (true) {
     const { doLogin } = yield race({
       doLogin: take(actionTypes.LOGIN_REQUESTED),
       doLogout: take(uiActionTypes.LOGOUT_REQUESTED),
     });
 
+    if (lastTask) {
```
Comment (efjacobson) [inline src/plugins/iam/sagas/index.js:20]: @hunyoboy might call this `loginTask` and only check to see if we need to cancel it in the `if (doLogin)`block

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # CHANGELOG
 
 
+## vN.N.N
+* issue #1238: cms will crash after session/raven disconnect and users accept new disconnect Alert
```
Comment (efjacobson) [inline CHANGELOG.md:5]: @hunyoboy remember we dont link to tickets that are not part of this repo

**Review by efjacobson — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (hunyoboy) [reply CHANGELOG.md]: @efjacobson removed the link and updated changed log.

File: src/plugins/iam/sagas/index.js (hunk 1 above)
Comment (hunyoboy) [reply src/plugins/iam/sagas/index.js:20]: @efjacobson thank you i got you but i think we also need to cancel if for some reason a logout request will be issued. if for some reason the login flow is still running ( auth server still pending, etc. ) and suddenly we want to logout, i think we really need to terminate the login flow to keep the state in sync.

**Comment by efjacobson** [comment]
this looks good to me, what do you think @gdbrown ?

**Review by efjacobson — APPROVED** [review]

**Review by gdbrown — CHANGES_REQUESTED** [review]
@hunyoboy can we also put a try catch around the raven connect and simply return if it fails.

https://github.com/triniti/cms-js/blob/1238-login-flow-issue/src/plugins/raven/services/Raven.js#L79

i believe that was also causing the white screen.

**Review by gdbrown — APPROVED** [review]

**Pull request merged**